### DISCUSSION TITLE
How do you make a codelist for high cost drugs?

### DISCUSSION THREAD

**Discussion by brianmackenna** [opensafely/documentation#1362] [Codelists]
I want to make a codelist for [high cost drugs](https://docs.opensafely.org/data-sources/hcd/)

There is no option on OpenCodelists and I can't find anything in the docs.

**Comment by brianmackenna** [comment]
High cost drugs file jas not seen much use due to the one off nature of the import. Consequently we made a whole pile of codelists at the start but never documented it.

@rc16 has spotted that its not entirely clear how to make them.

Opening this question to document how we make them as myself and @rc16 refresh our memory! Eventually this should make its way into docs

This issue has some clear concise descriptions
https://github.com/opensafely/immunosuppressant-meds-research/issues/36

but you will also need to follow the links to 
1. https://github.com/opensafely/immunosuppressant-meds-research/issues/29
2. https://github.com/opensafely/highcostdrugs-research/blob/master/analysis/unique_drug_names.R

still a Work in Progress to see if we get a final codelist!

**Comment by rc16** [comment]
I think I need to use the unique_drug_names.csv file outside the OS environment to create a codelist. I'm not clear if this file ever gets updated? If so, how to recreate this file?

I also don't have the option to create a high cost drug codelist on OpenCodelists - I presume that a csv with the drug names needs to be created and uploaded?

I understand that the field for the drug name searches for exact matches. I am wondering whether being able  to search for partial matches would make it much easier to query the high cost drugs - but I'm not sure if the tech team would want to do this or if it is the best thing to do but thought I'd put it on here to discuss?

**Comment by Jongmassey** [comment]
If you could create a Psuedo-BNF or dm+d codelist with these drugs that included the name, then I can see a way that that would work in ehrql and cohort-extractor if you used the name column as the "code" column

**Reply by brianmackenna** [reply]
Just to add some more context on this for future readers........................

PseudoBNF codes aren't optimal for this job and it relates to how BNF codes are made. Briefly High Cost Drugs relate to hospital medicines. BNF codes is a code system that arises from the reimbursement system for medicines prescribed by GPs. A BNF code is not made in advance of a medicine getting prescribed  - it only gets made once someone has prescribed it and it need to get reimbursed. Therefore only primary carer medicines will have a pseudoBNF code.

Whilst some hospital medicines do have BNF codes as they have been prescribed at least one point in primary care some (and particularly brands) will not so therefore don't have a BNF code.

On dm+d - these get made in advance BUT we have no tooling yet for a human user to compose long codelists natively in dm+d.

This blog has more onward links for further reading

https://www.bennett.ox.ac.uk/blog/2022/11/difference-between-bnf-dm-d-and-snomed-ct-codes/

**Comment by rose-higgins** [comment]
Oh i wrote [this doc](https://docs.google.com/document/d/12KJ5IWAkkcqfhS40St2CjUbOmh-PKHGyaT9QKfNh2EI/edit#heading=h.l93gy3w090ku) shortly after i first joined about making code lists from the High Cost Drugs data... I haven't touched it in ages, and it isn't the slickest process but definitely worked _back in the day_

**Comment by rc16** [comment]
Thanks @rose-higgins that's really helpful and I think does answer some of my questions. I'll try following this and post here if I have questions.

**Comment by rc16** [comment]
Hi,

I have been able to follow the document and have created the codelists locally. A couple of points:

- I found the drug names to search through the BNF - medicinal forms section. I did find that a few of the brand names did not have all of the sections on openprescribing but I added the brand name in the old_drug_name column. It did pick up a few extra records.
- For one drug obeticholic acid the vtm was a long number so got cut off when saved even though I saved it as text. But it still generated a codelist so hopefully this is not a problem?
- It doesn't say the next steps. I assume the codelists need adding to OpenCodelists? So that it can work on the real data? Then add the codelists using the codelists python script? This could do with some clarification. Also, I do not have the facility to add a high cost codelist so I think I need to ask tech-support.

**Reply by iaindillingham** [reply]
You need not add your codelist to OpenCodelists. Instead, commit it to your repository's `codelists` directory.

**Reply by brianmackenna** [reply]
I think thats ok to get the study up and running quickly but routinely we'd like to store them on OpenCodelists for re-use?

We have a whole suite of them already e.g. https://www.opencodelists.org/codelist/opensafely/high-cost-drugs-adalimumab/6eec91b5/

**Reply by rc16** [reply]
@iaindillingham if I commit it the the repository I assume I need to refer to it in codelists.py? If so which system should I use? example code:
oca_hcd_codes = codelist_from_csv(
  "codelists/hcd-obeticholic-acid-drug-names.csv",
  system = 
  column = "olddrugname",
)

**Reply by iaindillingham** [reply]
I'm afraid I don't know which system you should use. Perhaps @rose-higgins can help?

**Reply by Jongmassey** [reply]
If you are using cohort extractor then I _think_ you could use `system = None`

**Reply by rose-higgins** [reply]
Not much help but in the original project we ended up not actually using the HCD codelists so I didn't implement them.

**Reply by rc16** [reply]
Using system = none does seem to work - at least locally. When I upload the hcd codelists to GitHub the actions doesn't like it and says I need to run codelists update but if I do this it removes the hcd codelists. 

However, I then tried running [it](https://github.com/opensafely/udca_covid/blob/main/analysis/study_definition_hc_drugs.py) on the server and it does run, however it doesn't identify any prescriptions, whereas when I just used the drug name to identify high cost drugs [here](https://github.com/opensafely/udca_covid/blob/main/analysis/study_definition_pbc.py) it did identify people using the drug.

I assumed that the drugs names was correct as it hasn't been updated according to the page that the document links to. I'm not sure what the problem is..

**Reply by Jongmassey** [reply]
>but if I do this it removes the hcd codelists.

Ah yes it might do that. Try making a `local_codelists` folder and putting it in there, it shouldn't be touched by `codelists update` in there

**Reply by iaindillingham** [reply]
Sorry about that. Dave pointed out this behaviour, and I forgot to update my suggestion. Jon is correct: make a new folder.

**Comment by rc16** [comment]
Hi, I have one small point about creating the high cost codelist. When I searched for ursodeoxycholic acid some brand names did not come up in the virtual medicinal products. I added some extra brand names, that I found through medicines.org. It didn't result in a lot of extra codes but did pick up a few. For example in the picture below the old_drug_names ursofalk, cholurso and ursonorm were other brand names. 
![image](https://github.com/opensafely/documentation/assets/20856010/9e67b900-d0b0-41db-b3eb-c5e3f8442d25)

I just wanted to check this seemed like a reasonable thing to do before I updated the google doc?

**Reply by rc16** [reply]
Just to note I have made updates to the [document](https://docs.google.com/document/d/12KJ5IWAkkcqfhS40St2CjUbOmh-PKHGyaT9QKfNh2EI/edit), just await clarification to this question.
